src/screens/AddOrEditPost/selectImage.ts

```diff
@@ -0,0 +1,46 @@
+import { PermissionsAndroid, Platform } from "react-native";
+import ImagePicker from "react-native-image-crop-picker";
+import ImageResizer from "@bam.tech/react-native-image-resizer";
+
+const resizeImage = (path: string, setImageURL: (url: string) => void) => {
+  ImageResizer.createResizedImage(path, 1000, 1000, "JPEG", 10)
+    .then((response) => {
+      let pathAfterResize =
+        Platform.OS === "android" ? `file://${response.path}` : response.path;
+      setImageURL(pathAfterResize ?? "");
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+};
+
+const openImagePicke = (setImageURL: (url: string) => void) => {
+  ImagePicker.openPicker({
+    mediaType: "photo",
+    cropping: true,
+    freeStyleCropEnabled: true,
+  }).then((image) => {
+    resizeImage(image.path, setImageURL);
+  });
+};
+
+const androidPermission = async (setImageURL: (url: string) => void) => {
+  try {
+    const granted = await PermissionsAndroid.request(
+      PermissionsAndroid.PERMISSIONS.READ_EXTERNAL_STORAGE,
+    );
+    if (granted === PermissionsAndroid.RESULTS.GRANTED) {
+      openImagePicke(setImageURL);
+    } else {
+      console.log("error on openImagePicker ");
+    }
```

Comment:
Vad händer om man de inte ger Permission? Stängs alla dialoger och du är kvar på sidan?

Reply:
If user pressed permission denied and then try to add photo again, then comes alert with this text 

"Åtkomst nekad",
 "Vi behöver denna behörighet för att använda foto. Vänligen ge det i appinställningarna.",
"Cancel."      "Open Settings"


if user press    "Open Settings" then it will navigate to settings where user can give permission 

![Simulator Screenshot - iPhone SE (3rd generation) - 2024-05-08 at 10 07 51](https://github.com/Do-Good-Get-Good/do-good-get-good/assets/71877080/1c3d6a78-18d3-4677-89e8-8da2cdc57a80)
![Simulator Screenshot - iPhone SE (3rd generation) - 2024-05-08 at 09 58 01](https://github.com/Do-Good-Get-Good/do-good-get-good/assets/71877080/368dce46-6413-4969-9d56-e705a5ceff94)
![Simulator Screenshot - iPhone SE (3rd generation) - 2024-05-08 at 09 58 01](https://github.com/Do-Good-Get-Good/do-good-get-good/assets/71877080/ea720309-096c-4794-a968-380d6893b67b)